src/libraries/System.Data.Odbc/src/System/Data/Odbc/OdbcDataReader.cs

```diff
@@ -407,7 +407,10 @@ public override string GetDataTypeName(int i)
                 {
                     info._typename = GetColAttributeStr(i, ODBC32.SQL_DESC.TYPE_NAME, ODBC32.SQL_COLUMN.TYPE_NAME, ODBC32.HANDLER.THROW);
                 }
+// TODO: Behavior change probably needed here - when there's no data type, we mostly probably need to throw instead of returning null.
```

Comment:
```suggestion
// TODO-NULLABLE: Behavior change probably needed here - when there's no data type, we mostly probably need to throw instead of returning null.
```